### PR TITLE
Znika feature sidebar

### PR DESCRIPTION
**Pull request by jstkiddin**
9k+ - it's with yarn and ect

### CODE REVIEW

File: src/components/Navigation/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react'
+// import {useNavigate} from 'react-router-dom'
+import {CssBaseline,Toolbar, Typography, AppBar  } from '@mui/material'
+// import clsx from 'clsx'
+
+import UserBar from '../UserBarLogedOut'
+import UserBarLoggedIn from '../userBarLoggedIn'
+import useStyles from '../styles/useStyle'
+
+// function ifLoddedIn(){
```
Comment (VolodymyrDevico) [inline src/components/Navigation/NavBar.tsx:10]: remove commented stuff

File: src/components/Navigation/SideBarItem.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    const classes = useStyles()
+
+    const [list_item, setState] = useState(
+      {
+        activeButton: {},
+        objects:[
+          {
+            id:1,
+            text:'Upcoming events',
+            icon:<Campaign />,
```
Comment (VolodymyrDevico) [inline src/components/Navigation/SideBarItem.tsx:18]: Icon: Campaign 

 <Icon />

File: src/components/Navigation/SideBarItem.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            id:8,
+            text:'Privasy & Terms of use',
+            icon:<ScatterPlot />,
+            path:'/ptou'
+          }
+        ]
+      }
+    )
+
+    function toggleButton(index:any){
```
Comment (VolodymyrDevico) [inline src/components/Navigation/SideBarItem.tsx:66]: useCallback
And don't use any

**Review by VolodymyrDevico — CHANGES_REQUESTED** [review]

File: src/components/DynamicIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+function upperFirst(string:string) {
+    return string.slice(0, 1).toUpperCase() + string.slice(1, string.length)
+}
+  
+function fixIconNames(string:string) {
+    const name = string.split('-').map(upperFirst).join('')
+    if (name === '3dRotation') {
+        return 'ThreeDRotation'
+    } else if (name === '4k') {
```
Comment (VolodymyrDevico) [inline src/components/DynamicIcon.tsx:19]: necessary else

File: src/components/Navigation/NavBar/NotificationButton.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import NotificationDropDown from './NotificationDropDown'
+import clsx from 'clsx'
+
+const NotificationButton: React.FC = () =>{
+  const classes = useStyles()
+  const [invisible, setInvisible] = useState(false)
+  const [anchorEl, setAnchorEl] = useState(null)
+  const notifDropDownRef = useRef(null)
+
+  const handleOpenMenu = (e:any) =>{
```
Comment (VolodymyrDevico) [inline src/components/Navigation/NavBar/NotificationButton.tsx:14]: use callback and remove any type

File: src/components/Navigation/NavBar/NotificationDropDown.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+  return(
+    <>
+      <Menu ref={props.notifDropDownRef} 
+            onClose={props.handleCloseMenu}
+            anchorEl={props.anchorEl} 
+            open={Boolean(props.anchorEl)}
+            className={classes.userBarNotif}
+            >
+       {/* <ArrowDropUp className={clsx(classes.userBarComponentW, classes.userBarNotifArrow)} fontSize='large'/>       */}
```
Comment (VolodymyrDevico) [inline src/components/Navigation/NavBar/NotificationDropDown.tsx:34]: remove

File: src/components/Navigation/NavBar/dropDownMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        props.setLogged(!props.logged)
+        fillIfLogged
+        break
+      default:
+        break
+    }
+  }
+
+  const fillIfLogged = ()=>{
+    return props.logged?setMenu(menuIn):setMenu(menuOut)
```
Comment (VolodymyrDevico) [inline src/components/Navigation/NavBar/dropDownMenu.tsx:45]: if else better to call signature

File: src/components/Navigation/NavBar/dropDownMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        anchorEl={props.anchorEl} 
+        open={Boolean(props.anchorEl)}
+        className={classes.userBarDropdown}
+        >
+        {menu.map((item,index)=>(
+          <MenuItem
+            component='button' 
+            key={index}
+            className={classes.userBarDropdownButtons}
+            onClick={()=>handleClick(item.name)}
```
Comment (VolodymyrDevico) [inline src/components/Navigation/NavBar/dropDownMenu.tsx:62]: don't use arrow functions at render

**Review by VolodymyrDevico — CHANGES_REQUESTED** [review]

File: src/components/Navigation/NavBar/userBarLoggedIn.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+
+
+const UserBarLoggedIn: React.FC = () =>{
+  const classes = useStyles()
+  const [invisible, setInvisible] = useState(false)
+
+  // const handleBadgeVisibility = () => {
+  //   setInvisible(!invisible)
+  // }
```
Comment (ntnshpig) [inline src/components/Navigation/NavBar/userBarLoggedIn.tsx:19]: don`t leave comments

**Review by ntnshpig — CHANGES_REQUESTED** [review]

**Pull request closed without merging**